tools/2025-03-workflow-changes.md

```diff
@@ -0,0 +1,260 @@
+
+Some of the existing Make commands invoke scripts. This pattern will continue in order to keep all significant logic isolated to scripts because we do not want to move the complexity problem from GitHub workflows into Make. Simple composite commands should remain within the Make files as they are today.
+
+#### Configuration is provided through environment variables
+
+Environment variables can be set on developer machines (and stored in `.env` files). GitHub workflows will set environment variables during setup steps. Core logic will read the environment variables.
+
+#### Workflow schedules do not trigger on forks
+
+Some workflows have scheduled executions. It does not make sense for these schedules to be triggered on forks; running scheduled workflows on forks only wastes compute time of the runners.
+
+#### All workflows can run on dispatch and run on branches
+
+All workflows can be manually triggered, in addition to the other ways they can be triggered, like on pull requests and on a schedule. Allowing manual triggers enables testing from forks.
+
+All workflows can be run on branches so that fork owners do not have to merge the changes to the main branch of the fork. Having to run forks from the main branch makes fork synchronization more difficult.
+
+#### Workflows that perform PR checks should be configurable to run from the committing branch
+
+When a PR includes a change to a workflow, the workflow should be runnable from the branch if the person running the workflow has write access to the repo. (PRs submitted from forks where the contributor does not have write access should continue to require approval to run by someone with write access.)
+
+### Design options considered but not chosen
+
+* Adopt a new automation tool like [Just](https://github.com/casey/just) or [Task](https://taskfile.dev/), and deprecate Make. These tools have advantages, but a side-effect of adopting a new tool would likely be the existence of a new tool in our toolbox without the removal of Make, which would increase complexity. Make currently meets our needs for invoking developer automation.
+* Remove everything from GitHub workflows except the invocation of another tool: GitHub workflows have advantages when setting up the automation environment, e.g. retrieving credentials, cloning repos, configuring runners, installing tools, etc.
+
+## Security
+
+### Least Privilege
+
+Workflows will continue to adhere to the least privilege feature in which they declare which permissions are required by the workflow.
+
+### Identity
+
+Where possible, the automation logic will use the GitHub CLI, which has identity built in. Developers would log into the CLI when running code on their machines, which would allow them to run the logic on their forks. When running as part of at GitHub workflow, the GitHub CLI context is already set up and no further configuration is required.
```

Comment:
Same question as above, is this different from what Radius workflows do today?

Reply:
I believe there are some instances where git commands are used where a GH CLI command would be simpler, but this is an overall principle rather than a change to a specific instance in the current workflows.